src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,257 @@
+        # If there are multiple fields, combine metrics for each field into a single dataframe
+        final_metrics = pd.concat(field_metrics, axis=1, keys=field_metrics.keys())
+        return final_metrics
+
+    def _clean_column_name(self, col_name: str) -> str:
+        """
+        Clean the column name of a dataframe to fit a specific regex pattern.
+
+        Args:
+            col_name (str): A column name.
+
+        Returns:
+            str: A column name that fits the regex pattern.
+        """
+        col_name = col_name.lower()  # Convert to lowercase
+        col_name = re.sub(r"[^a-z0-9_]", "_", col_name)  # Replace non-alphanumeric characters with underscores
+        return col_name
+
+    def _cast_to_python_types(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Convert integer, boolean and floats columns in a dataframe
+        to Python native types.
+
+        Args:
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            pd.DataFrame: The text descriptives dataframe with integer and boolean columns cast to Python native types.
+        """
+        # Select columns by data type
+        int_cols = df.select_dtypes(include=["int64"]).columns
+        bool_cols = df.select_dtypes(include=["boolean"]).columns
+        float_cols = df.select_dtypes(include=["float64"]).columns
+        # Cast integer columns to Python's native int type
+        df[int_cols] = df[int_cols].astype(int)
+        # Cast boolean columns to Python's native str type
+        df[bool_cols] = df[bool_cols].astype(str)
+        # Cast float columns to Python's native float type and round to 2 decimal places
+        df[float_cols] = df[float_cols].astype(float).round(2)
+        return df
+
+    def _create_metadata_properties(self, df: pd.DataFrame) -> List:
+        """
+        Generate metadata properties based on dataframe columns and data types.
+
+        Args:
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            List: A list of metadata properties.
+        """
+        ### TO DO: Handle nans
+        properties = []
+        for col, dtype in df.dtypes.items():
+            name = f"{self._clean_column_name(col)}"
+            title = name.replace("_", " ").title()
+            if dtype == "object":
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            elif dtype == "int64":
+                prop = IntegerMetadataProperty(
+                    name=name, title=title, visible_for_annotators=self.visible_for_annotators
+                )
+            elif dtype == "float64":
+                prop = FloatMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            elif dtype == "bool":
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
```

Comment:
Why is `dtype == "object"` and `dtype == "bool"` both interpreted as `TermsMetadataProperty`?

And if that's the case, wouldn't it be better to just combine both within the same if-statement?

```suggestion
            if dtype in ["object", "bool"]:
                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
            elif dtype == "int64":
                prop = IntegerMetadataProperty(
                    name=name, title=title, visible_for_annotators=self.visible_for_annotators
                )
            elif dtype == "float64":
                prop = FloatMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
```

Reply:
Also, don't we know all the potential values within a column? e.g. all the unique terms in a column?

If so, we could add those to the property e.g. `TermsMetadataProperty(..., terms=[...], ...)`

Reply:
As far as I can tell from the [documentation](https://github.com/argilla-io/argilla/blob/e270f8788f6668e194dbedb5f9086c2beab6c247/src/argilla/client/feedback/schemas/metadata.py#L106), there is no metadata property for boolean values, which is why `dtype == "object"` and `dtype == "bool"` are both covered by the `TermsMetadataProperty`.

Reply:
And yes, I just looked up the text-descriptives documentation and found that the only non-integer or non-float values are actually just True/False, so I also added `values=["True", "False"]`.